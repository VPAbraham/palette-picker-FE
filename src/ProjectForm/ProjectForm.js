import React, {Component} from 'react';
import './ProjectForm.scss';
import whitePlus from '../assets/images/plus_white.svg';


class ProjectForm extends Component {
  constructor(props) {
    super();
    this.state = {
      creatingProj: false
    }
  }

  render() {
    return(
      <section className="project-form">
        <div className="menu-items">
          <img src={whitePlus}
            onClick={() => this.setState({ creatingProj: !this.state.creatingProj })}
            alt="white plus symbol"
            className="modalPlus"
          />
          <h1>CREATE NEW PROJECT</h1>
        </div>
        {this.state.creatingProj &&
        <div className="proj-name-input">
          <input className="proj-name-input" placeholder="New Project Name"></input>
          <button>SUBMIT</button>
        </div>
        }
        <div className="menu-items">
          <img src={whitePlus} alt="white plus symbol" className="modalPlus" />
          <h1>VIEW ALL PROJECTS</h1>
        </div>
      </section>
    )
  }
}

export default ProjectForm;