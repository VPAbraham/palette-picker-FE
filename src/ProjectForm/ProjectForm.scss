.project-dropdown {
  margin-left: 3em;
}
