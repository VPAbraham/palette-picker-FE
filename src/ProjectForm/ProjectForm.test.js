import React from 'react';
import { ProjectForm } from './ProjectForm';
import { shallow } from 'enzyme';

describe('ProjectForm', () => {
  let wrapper;
  beforeEach(() => {
    const projects = [{ id: 20, name: 'Fall Colors' }, { id: 21, name: 'Ocean Vibes' }, { id: 22, name: 'Seeing Red' }];
    const palettes = [
      { id: 6, name: 'Pal1', projects_id: 20, color1: '#baddc3', color2: '#ff6767', color3: '#ffe596', color4: '#b3e6ff', color5: '#632ad7' },
      { id: 7, name: 'Pal2', projects_id: 20, color1: '#ab7146', color2: '#b3e6ff', color3: '#41406d', color4: '#b3e66tr', color5: '#41406d' },
      { id: 8, name: 'Pal3', projects_id: 22, color1: '#993f42', color2: '#d8ac97', color3: '#4e6cc4', color4: '#b4r35', color5: '#41406d' }
    ]
    wrapper = shallow(<ProjectForm
      projects={projects}
      palettes={palettes}
      selectPalette={jest.fn()}
      hover={false}
    />)
  });
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});