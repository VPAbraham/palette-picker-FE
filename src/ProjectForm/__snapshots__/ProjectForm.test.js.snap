// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm should match the snapshot with all data passed in 1`] = `
<section
  className="project-form"
>
  <div
    className="menu-items"
  >
    <img
      alt="white plus symbol"
      className="modalPlus"
      onClick={[Function]}
      src="plus_white.svg"
    />
    <h1>
      CREATE NEW PROJECT
    </h1>
  </div>
  <div
    className="menu-items"
  >
    <img
      alt="white plus symbol"
      className="modalPlus"
      onClick={[Function]}
      src="plus_white.svg"
    />
    <h1>
      VIEW ALL PROJECTS
    </h1>
  </div>
</section>
`;
